src/components/HistoryList/items/TokenTransactionItem.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+
+// Components
+import TokenValueView from 'components/modern/TokenValueView';
+
+// Selectors
+import { useRootSelector } from 'selectors';
+
+// Utils
+import { findEnsNameCaseInsensitive } from 'utils/common';
+import { formatHexAddress } from 'utils/format';
+import { useThemeColors } from 'utils/themes';
+
+// Types
+import type { TokenReceivedEvent, TokenSentEvent } from 'models/History';
+
+// Local
+import HistoryListItem from './HistoryListItem';
+
+
+type Props = {|
+  event: TokenReceivedEvent | TokenSentEvent,
+  onPress?: () => mixed,
+|};
+
+function TokenTransactionItem({ event, onPress }: Props) {
+  const colors = useThemeColors();
+
+  const ensRegistry = useRootSelector((root) => root.ensRegistry.data);
+
+  if (event.type === 'tokenReceived') {
+    const ensName = findEnsNameCaseInsensitive(ensRegistry, event.fromAddress);
+
+    return (
+      <HistoryListItem
+        iconName="arrow-down"
+        iconColor={colors.positive}
+        iconBorderColor={colors.positiveWeak}
+        title={ensName ?? formatHexAddress(event.fromAddress)}
+        valueComponent={
+          <TokenValueView value={event.value.value} symbol={event.value.symbol} variant="medium" mode="change" />
```

Comment:
I wonder if we can add helper in the above for this (and other items possibly) to destruct `value`, wdyt? Also it looks like mistype, on the helper it could be mentioned that this is intended, wdyt?

Reply:
I've added destructuring, indeed it improved readability.

I'm not sure though what do you exactly mean by "mistype" here? Pls elaborate.

Reply:
For the same thing (double value key), but I guess if you destructed it should be fine 👍

Reply:
Yeah naming here is tricky, so the type is `TokenValue` which is `value: BigNumber` + `symbol: string`, which kinda seems alright, but then you have `TokenReceivedEvent` with `value: TokenValue` field, and then it get's a bit awkward, but I guess this is quite minor and mostly aesthetic thing